source/extensions/filters/http/ratelimit/ratelimit.cc

```diff
@@ -273,6 +274,34 @@ VhRateLimitOptions Filter::getVirtualHostRateLimitOption(const Router::RouteCons
   return vh_rate_limits_;
 }
 
+const Router::RateLimitPolicy&
+Filter::getRouteRateLimitPolicy(const Router::RouteConstSharedPtr& route) {
+  const FilterConfigPerRoute* specific_per_route_config;
+
+  if (route->routeEntry()->rateLimitPolicy().empty()) {
+    specific_per_route_config = dynamic_cast<const FilterConfigPerRoute*>(
```

Comment:
```
Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfigPerRoute>(
            HttpFilterNames::get().RateLimit, route); 
```
instead of dynamic_cast? Or I don't understand something?